src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,148 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    static class Node<T>{
+        T item;
+        Node<T> prev;
+        Node<T> next;
+
+        public Node(T item) {
+            this.item = item;
+        }
+    }
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value);
+        if (size == 0){
+            head = tail = newNode;
+        } else {
+            tail.next = newNode;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new ArrayIndexOutOfBoundsException();
+        }
+        Node<T> newNode = new Node<>(value);
+        if (head == null){
+            head = tail = newNode;
+        } else if (index == 0){
+            newNode.next = head;
+            head = newNode;
+        } else if (index == size){
+            newNode.prev = tail;
+            tail = newNode;
+            tail.next =newNode;
```

Comment:
In the `add(T value, int index)` method, when adding a node at the end, you should set `tail.next = newNode` instead of `tail = newNode; tail.next = newNode;` to correctly update the tail reference.